### DISCUSSION TITLE
How to setup Scheduled Task in nextjs project

### DISCUSSION THREAD

**Discussion by backurdi** [coollabsio/coolify#3228] [Q&A] [answered]
Hello people,
I am migrating my NextJS project from vercel to Coolify, Cron jobs in Vercel was pretty straight forward as I just create endpoints in a Cron folder in the api section of the project. Then I define each cronjob like this
{
  "crons": [
    {
      "path": "/api/cron",
      "schedule": "*/5 * * * *"
    },
    {
      "path": "/api/cron/funnelDrop",
      "schedule": "* * * * *"
    },
    {
      "path": "/api/cron/meetingReminder",
      "schedule": "* * * * *"
    },
    {
      "path": "/api/cron/inviteFollowUp",
      "schedule": "0 8 * * *"
    }
  ],
  "buildCommand": "sh vercel.sh"
}

How can I set it up in Coolify?

I tried using Curl to call the whole endpoint, but that did not work in Coolify.

Anyone who have experience with this?

**Comment by Vahor** [comment]
What have you tried ?
I'm using `Scheduled Task` with curl like you just said and it works.

For instance I have a task scheduled every hour with:
- Frequency: `0 * * * *`
- Command: `curl --fail -X POST http://localhost:3000/[endpoint]`

if you want to check that your command works you can use the Command tab

**Comment by backurdi** [comment]
<img width="1084" alt="Screenshot 2024-09-02 at 11 55 19" src="https://github.com/user-attachments/assets/165236fe-49bc-4019-b856-cdff3fda7ea5">

It looks like this, the endpoint I am trying to hit is  https://staging.capio.pro/api/cron, it works if I run curl from the terminal or the server itself but not as a cron job

**Reply by Vahor** [reply]
Do you have any error ?

**Reply by backurdi** [reply]
Just tested in the command tab. Got this error
sh: 1: curl: not found

**Reply by backurdi** [reply, marked as answer]
Got it to work by installing curl on build

**Reply by backurdi** [reply]
Thanks for the help 🫶
